File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count =1,res =0;
        if(nums.length==1){
            return nums[0];
        }
        for(int i =0;i<nums.length;i++){
            if(nums[i]==nums[res]){
                count++;
            }
            else{
                count--;
            }
            if(count==0){
                res = i;
                count =1;
            }
        }
        count =0;
        for(int i =0;i<nums.length;i++){
            if(nums[res]==nums[i]){
                count++;
            }
        }
        if(count<=nums.length/2){
            res =-1;
        }
        return nums[res];

    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # if len(nums)==1:
        #     return nums[0]
        # for i in range(len(nums)):
        #     count =1
        #     for j in range(i+1, len(nums)):
        #         if nums[i]==nums[j]:
        #             count+=1
        #         if count > len(nums)/2:
        #             return nums[i]
        # return -1
        x,count =None,0
        for num in nums:
            if count == 0:
                x = num
            if x == num:
                count+=1
            else :
                count-=1
        return x
